books/RayTracingTheNextWeek.html

```diff
@@ -2256,7 +2256,319 @@
+we say "quadrilateral" here, we're really talking about parallelograms, a specific kind of
+quadrilateral where opposite sides are parallel. These three entities are
+
+1. $\mathbf{O}$, the lower-left corner of the quadrilateral.
+2. $\mathbf{u}$, a vector representing the first side of the quadrilateral.
+   $\mathbf{O} + \mathbf{u}$ gives the lower right corner.
+3. $\mathbf{v}$, a vector representing the second side of the quadrilateral.
+   $\mathbf{O} + \mathbf{v}$ gives the upper left corner.
+
+These values are three-dimensional, even though a quad itself is a two-dimensional object.
+
+The following figure illustrates the quadrilateral components.
+
+  ![Figure [quad-def]: Quadrilateral Components](../images/fig-2.05-quad-def.jpg)
+
+
+Finding the Plane That Contains a Given Quadrilateral
+------------------------------------------------------
+We have $O$, $u$, and $v$, and want the corresponding equation of the plane containing all three of
+these values (and thus the quad itself).
+
+Fortunately, this is very simple. Recall that in $Ax + By + Cz = D$, $(A,B,C)$ represents the
+normal vector. To get this, we just use the cross product of the two side vectors $u$ and $v$:
+
+  $$ \mathbf{n} = \text{unit_vector}(\mathbf{u} \times \mathbf{v}) $$
+
+The plane is defined as all points $(x,y,z)$ that satisfy the equation $Ax + By + Cz = D$. Well, we
+know that $\mathbf{O}$ lies on the plane, so we can use that to find $D$:
+
+  $$ D = \mathbf{n}_x\mathbf{O}_x + \mathbf{n}_y\mathbf{O}_y + \mathbf{n}_z\mathbf{O}_z $$
+
+We can now use these two values $\mathbf{n}$ and $D$ to find the point of intersection with a given
+ray and the plane containing the quadrilateral.
+
+
+Orienting Points on The Plane
+------------------------------
+At this stage, the intersection point is on the plane that contains the quadrilateral, but it could
+be _anywhere_ on the plane, and may lie inside or outside the quadrilateral. We need to determine if
+the intersection lies inside (hits) the quadrilateral, and reject points that lie outside (miss) the
+quad. To determine where a point lies relative to the quad, and to assign texture coordinates to the
+point of intersection, we need to somehow orient the intersection point on the plane.
+
+To do this, we'll construct a _coordinate frame_ for the plane -- a way of orienting points located
+on the plane. We've already been using a coordinate frame for our 3D space -- this is defined by an
+origin point $\mathbf{O}$ and three basis vectors $\mathbf{x}$, $\mathbf{y}$, and $\mathbf{z}$.
+
+Since a plane is a 2D construct, we just need a plane origin point $\mathbf{O}$ and _two_ basis
+vectors: $\mathbf{u}$ and $\mathbf{v}$. Now normally, you see axes that are perpendicular to each
+other, so they all meet at 90° angles. However, this doesn't need to be the case in order to span
+the entire space -- you just can't have two axes that are parallel to each other.
```

Comment:
However, this doesn't need to be the case in order to span the entire space -- you just need two axes that are not parallel to each other.

Reply:
oopsie

Reply:
Better.